sequencer/src/persistence/sql.rs

```diff
@@ -1389,6 +1392,62 @@ impl SequencerPersistence for Persistence {
     }
 }
 
+#[async_trait]
+impl DhtPersistentStorage for Persistence {
+    /// Save the DHT to the database
+    ///
+    /// # Errors
+    /// - If we fail to serialize the records
+    /// - If we fail to write the serialized records to the DB
+    async fn save(&self, records: Vec<SerializableRecord>) -> anyhow::Result<()> {
+        // Bincode-serialize the records
+        let to_save =
+            bincode::serialize(&records).with_context(|| "failed to serialize records")?;
+
+        // Prepare the statement
+        let stmt = "UPSERT INTO libp2p_dht (id, serialized_records) VALUES (0, $1)";
```

Comment:
ah I meant upsert function on transaction. 
something like this [here](https://github.com/EspressoSystems/espresso-sequencer/blob/main/sequencer/src/persistence/sql.rs#L1178)